Assets/Scripts/Loot.cs

```diff
@@ -2,19 +2,78 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+[System.Serializable]
+public struct Loots {
+    public GameObject ressource;
+    public int number;
+}
+
 public class Loot : MonoBehaviour {
-    public GameObject[] loot;
-    private bool isQuitting;
+    public List<Item> tools;
+    public List<Loots> loots;
+    public int maxHealth;
+    public int stepNumber;
+
+    private int health;
+    private int step;
+    private int nextStep;
+    private int ressourcesByStep;
+
+    private void Start() {
+        health = maxHealth;
+        step = health / (stepNumber + 1);
+        nextStep = health - step;
 
-    private void OnApplicationQuit() {
-        isQuitting = true;
+        var totalRessources = 0;
+        foreach (var loot in loots) {
+            totalRessources += loot.number;
+        }
+
+        ressourcesByStep = totalRessources / (stepNumber + 1);
     }
 
-    private void OnDestroy() {
-        if (isQuitting) return;
-        foreach (var item in loot) {
-            var spawnPos = transform.position + (Vector3) Random.insideUnitCircle + new Vector3(0, 2, 0);
-            Instantiate(item, spawnPos, Quaternion.identity);
+    public void Extract(Item item, int damage) {
+        if (!tools.Contains(item)) {
+            Debug.Log("Wrong tool");
+            return;
+        }
+
+        health -= damage;
+        Debug.Log("Aie");
+
+        if (health <= 0) {
+            foreach (var loot in loots) {
+                for (var i = 0; i < loot.number; i++) {
+                    var spawnPos = transform.position + (Vector3)Random.insideUnitCircle + new Vector3(0, 2, 0);
+                    Instantiate(loot.ressource, spawnPos, Quaternion.identity);
+                }
+                Destroy(this.gameObject);
+            }
+        } else {
```

Comment:
I looked a long time at the code below and I still have no idea how it works, but it seems to work fine, so yey :)